src/test/java/perkmanager/LogInAndSignUpTests.java

```diff
@@ -0,0 +1,115 @@
+package perkmanager;
+
+import org.junit.After;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.transaction.annotation.Transactional;
+
+import static org.mockito.Mockito.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+public class LogInAndSignUpTests {
+
+    @Autowired
+    private MockMvc mockMvc;
+    @Autowired
+    private PersonRepository personRepository;
+
+
+    @Test
+    @WithMockUser
+    public void testLoginPage() throws Exception {
+        mockMvc.perform(get("/login"))
+                .andExpect(status().isOk())
+                .andExpect(view().name("login"));
+    }
+
+    @Test
+    @WithMockUser
+    public void testSuccessfulLogin() throws Exception {
+        mockMvc.perform(post("/login")
+                        .param("username", "mutaz")
+                        .param("password", "mutaz"))
+                .andExpect(status().is3xxRedirection())
+                .andExpect(redirectedUrl("/LandingPage")); // Change this to the URL you expect after successful login
+    }
+
+    @Test
+    @WithMockUser
+    public void testUnsuccessfulLogin() throws Exception {
```

Comment:
Can you split this up to test for incorrect username and another test to test with incorrect password?